tdp/core/deployment/deployment_iterator.py

```diff
@@ -106,115 +106,122 @@ def __init__(
         except NothingToReconfigureError:
             self._reconfigure_operations = None
 
-    def __next__(self) -> Optional[list[SCHStatusLogModel]]:
+    def __next__(self) -> Callable[[], Optional[list[SCHStatusLogModel]]]:
         try:
             while True:
                 operation_rec: OperationModel = next(self._iter)
+                operation_rec.state = OperationStateEnum.RUNNING
```

Comment:
This is incorrect: in case of failure, operation state will be `RUNNING` and then `HELD`. IMO it should be `HELD` directly.

Reply:
Before the execution of the operation the state is `PENDING`. Then when its starts it is `RUNNING`. The executor changes the state from `RUNNING` to `SUCCESS` or `FAILURE` and all operations which have not been executed pass from `PENDING` to `HELD` if the deployment fails as it is stated in line 117-119.

Reply:
No, we shouldn't even start the execution if the deployment is in `FAILURE`.